File: scripts/algorithms/M/Maximum Number of Weeks for Which You Can Work/Maximum Number of Weeks for Which You Can Work.java
// Runtime: 3 ms (Top 73.74%) | Memory: 55.50 MB (Top 96.97%)

class Solution {
    public long numberOfWeeks(int[] milestones) {
        long sum = 0;
        int max = Integer.MIN_VALUE;
        for(int milestone: milestones) {
            sum += milestone;
            max = Math.max(milestone, max);
        }
        if((sum - max) < max)
            return ((sum - max) * 2) + 1;
        else
            return sum;
    }
}

File: scripts/algorithms/M/Maximum Number of Weeks for Which You Can Work/Maximum Number of Weeks for Which You Can Work.py
# Runtime: 1032 ms (Top 59.60%) | Memory: 25.7 MB (Top 80.23%)
class Solution:
    def numberOfWeeks(self, m: List[int]) -> int:
        return min(sum(m), 2 * (sum(m) - max(m)) + 1)